python_modules/dagster/dagster/_core/definitions/module_loaders/object_list.py

```diff
@@ -70,6 +76,20 @@ def assets_defs(self) -> Sequence[AssetsDefinition]:
     def source_assets(self) -> Sequence[SourceAsset]:
         return [asset for asset in self.deduped_objects if isinstance(asset, SourceAsset)]
 
+    @cached_property
+    def schedule_defs(self) -> Sequence[ScheduleDefinitionObject]:
+        return [
+            asset for asset in self.deduped_objects if isinstance(asset, RuntimeScheduleObjectTypes)
```

Comment:
If we can't do this automatically then I would much prefer putting the types here explicitly rather than importing the type tuple from elsewhere.

So that would at least be replacing RuntimeScheduleObjectTypes with the explicit (ScheduleDefinition, UnresolvedPartitionedAssetScheduleDefinition)

I also think ScheduleDefinitionObject is confusing enough of a name that we should probably just put the Union[ScheduleDefinition, UnresolvedPartitionAssetScheduleDefinition] there as well.

This is actually not any more redundant than the current solution and localizes the redundancy so it's easier to wrap your head around (and doesn't require the creation of a new concept).

The only place where this adds more LoC is the LoadableDagsterObject bit (where we do need to re-list all of these types), which I think is acceptable.

I'd rather spend the lines of code and keep everything in one file.

Reply:
fair enough